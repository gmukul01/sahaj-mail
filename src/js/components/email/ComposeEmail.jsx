import React from 'react';

import Modal from 'components/Modal';
import { useFormInput } from 'effects/useFormInput';

const ComposeEmail = ({ showModal, onCloseModal, sendEmail }) => {
	const to = useFormInput(''),
		subject = useFormInput(''),
		body = useFormInput('');

	const onSend = () => {
		const email = {
			to: to.value,
			subject: subject.value,
			body: body.value
		};
		onCloseModal();
		sendEmail(email);
	};

	return (
		<Modal {...{ showModal, onCloseModal, onConfirmModal: onSend, confirmText: 'Send', header: 'New Message' }}>
			<div className="compose-email">
				<div className="email-field">
					<label>To</label>
					<input type="text" {...to} />
				</div>

				<div className="email-field">
					<label>Subject</label>
					<input type="text" {...subject} />
				</div>

				<div className="email-field">
					<label>Body</label>
					<input type="text" {...body} />
				</div>
			</div>
		</Modal>
	);
};

export default ComposeEmail;
