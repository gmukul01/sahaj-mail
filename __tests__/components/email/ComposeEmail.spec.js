import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import ComposeEmail from 'components/email/ComposeEmail';

describe('ComposeEmail Component', () => {
	const mockOnCloseModal = jest.fn();
	const mockSendEmail = jest.fn();
	const initialProps = {
		showModal: false,
		onCloseModal: mockOnCloseModal,
		sendEmail: mockSendEmail
	};

	it('should match snapshot', () => {
		const { container } = render(<ComposeEmail {...initialProps} />);
		expect(container).toMatchSnapshot();
	});

	it('should show error when all the fields are empty', () => {
		const { container, getByText } = render(<ComposeEmail {...initialProps} />);
		fireEvent.click(getByText('Send'));
		expect(container.querySelector('.error').textContent).toBe('Please enter details');
	});

	it('should show error when body field is empty', () => {
		const { container, getByPlaceholderText, getByText } = render(<ComposeEmail {...initialProps} />);
		fireEvent.change(getByPlaceholderText('Enter recipient email id'), { target: { value: 'Dummyemail' } });
		fireEvent.click(getByText('Send'));
		expect(container.querySelector('.error').textContent).toBe('Please enter message');
	});

	it('should show error when email field is empty', () => {
		const { container, getByPlaceholderText, getByText } = render(<ComposeEmail {...initialProps} />);
		fireEvent.change(getByPlaceholderText('Enter message'), { target: { value: 'Dummy message' } });
		fireEvent.click(getByText('Send'));
		expect(container.querySelector('.error').textContent).toBe('Please enter recipient email id');
	});

	it('should call fetch Username use details if email and password is provided ', () => {
		const { getByPlaceholderText, getByText } = render(<ComposeEmail {...initialProps} />);
		fireEvent.change(getByPlaceholderText('Enter recipient email id'), { target: { value: 'Dummy Email' } });
		fireEvent.change(getByPlaceholderText('Enter subject'), { target: { value: 'Dummy Subject' } });
		fireEvent.change(getByPlaceholderText('Enter message'), { target: { value: 'Dummy message' } });
		fireEvent.click(getByText('Send'));
		expect(mockSendEmail.mock.calls[0][0]).toEqual({ to: 'Dummy Email', subject: 'Dummy Subject', body: 'Dummy message' });
	});
});
