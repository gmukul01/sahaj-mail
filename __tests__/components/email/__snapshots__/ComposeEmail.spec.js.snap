// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComposeEmail Component should match snapshot 1`] = `
<div>
  <div
    class="modal"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <h3
          class="title"
        >
          New Message
        </h3>
        <button
          class="modal-close"
        >
          Ã—
        </button>
      </div>
      <div
        class="modal-body"
      >
        <form
          class="compose-email"
        >
          <label>
            To
          </label>
          <input
            placeholder="Enter recipient email id"
            type="text"
            value=""
          />
          <label>
            Subject
          </label>
          <input
            placeholder="Enter subject"
            type="text"
            value=""
          />
          <label>
            Body
          </label>
          <textarea
            placeholder="Enter message"
          />
          <button
            class="button-solid"
            type="submit"
          >
            Send
          </button>
        </form>
        
      </div>
    </div>
  </div>
</div>
`;
